app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,81 @@
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Class for providing jwt identity from the decoded token from the field specified in a secret
+      class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+        def initialize(
+          authentication_parameters,
+          fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+          resource_class: ::Resource,
+          logger: Rails.logger
+        )
+          @logger = logger
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+          @authentication_parameters = authentication_parameters
+        end
+
+        def jwt_identity
+          return @jwt_identity if @jwt_identity
+
+          token_field_name = fetch_token_field_name
```

Comment:
this line can be removed if you change the method name from `fetch_token_field_name` to `token_field_name`

Reply:
Done